index.bs

```diff
@@ -8508,6 +8527,357 @@ worker=] algorithm:
+    Can be set to either <code>[=Lax=]</code> or <code>[=Strict=]</code>.
+ </tr>
+</table>
+
+<div algorithm>
+To <dfn>match cookie</dfn> given |stored cookie| and |filter|:
+
+1. For each |name| → |value| in |filter|:
+  1. If |name| is "<code>value</code>":
+    1. Set |value| to [=deserialize protocol bytes=] with |value|.
+  1. Let |field name| be the field name corresponding to the JSON key |name| in
+    the [=table for cookie conversion=].
+  1. If |stored cookie|[|field name|] does not equal |value|:
+    1. Return false.
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn>get matching cookies</dfn> given |cookie store| and |filter|:
+
+1. Let |cookies| be a new list.
+1. For each |stored cookie| in |cookie store|:
+  1. If [=match cookie=] with |stored cookie| and |filter| is true:
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+1. Return |cookies|.
+
+</div>
+
+### Commands ### {#module-storage-commands}
+
+#### The storage.getCookies Command ####  {#command-storage-getCookies}
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = {
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        }
+
+        storage.CookieFilter = {
+          ? ~ network.Cookie
+        }
+
+        storage.GetCookiesParameters = {
+          cookie: storage.CookieFilter,
+          ? partitionKey: storage.PartitionKey,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.GetCookiesResult = {
+        cookies: [*network.Cookie]
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for script.getCookies">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |filter| be the value of the <code>cookie</code> field of |command parameters|.
+1. Let |partition key spec| be the value of the <code>partitionKey</code> field of
+  |command parameters| if it is present or null if it isn't.
+1. Let |partition key| be the result of [=expand a storage partition key spec|expanding=] |partition key spec|.
+1. Let |store| be the result of [=trying=] to [=get the cookie store=] with
+  |partition key|.
+1. Let |cookies| be the result of [=get matching cookies=] with |store| and |filter|.
+1. Let |body| be a [=/map=] matching the <code>storage.GetCookiesResult</code> production,
+  with the <code>cookies</code> field set to |cookies|.
+1. Return [=success=] with data |body|.
+
+</div>
+
+#### The storage.setCookie Command ####  {#command-storage-setCookie}
+
+The <dfn export for=commands>storage.setCookie</dfn> command creates a new [=cookie=] in a cookie store, replacing any cookie in that store which matches according to [[COOKIES]].
```

Comment:
A command like `storage.setCookies` would involve a variable number of fallible operations. That might mean aborting on the first failure, or it might mean attempting all operations regardless of individual failures. I can see the value in both, so it seemed prudent to instead specify `storage.setCookie` and allow consumers to build their preference on top of that.

Reply:
You're right, @jgraham, I'm not fond of it, but as it duplicates existing Classic behavior, I'm not going to kick about it. Round-trips are perf killers in high-latency scenarios.